util/data_object.py

```diff
@@ -24,6 +24,39 @@ def exists(ctx, path):
                genquery.AS_LIST, ctx))) > 0
 
 
+def get_properties(ctx, data_id, resource):
+    """ Retrieves default properties of a data object from iRODS.
+
+    :param ctx:                                   Combined type of a callback and rei struct
+    :param data_id:                               data_id of the data object
+    :param resource:                              Name of resource
+
+    :returns: dictionary mapping each requested property to its retrieved value, or None for each value if not found.
+    """
+    # Default properties available for retrieva
+    properties = [
+        "DATA_ID", "DATA_MODIFY_TIME", "DATA_OWNER_NAME", "DATA_SIZE",
+        "COLL_ID", "DATA_RESC_HIER", "DATA_NAME", "COLL_NAME",
+    ]
+
+    # Retrieve data obejct with default properties
+    query_fields = ", ".join(properties)
+    iter = genquery.row_iterator(
+        query_fields,
+        "DATA_ID = '{}' AND DATA_RESC_HIER like '{}%'".format(data_id, resource),
+        genquery.AS_LIST, ctx
+    )
+
+    # Return a None dict when no data object is found
+    prop_dict = {prop: None for prop in properties}
```

Comment:
I made small change based on what we discussed. In case data object is not found, returns None dictionary rather than a None. This is because returning None will raise an error to stop the revision process (so not skipping).

Reply:
I think you can check the return value in `revision_create`, right?

I'm thinking of something along the lines of:
```
if revision_properties is None:
      log.write(ctx, "Not creating revision for data object with ID {}, because it no longer exists on resource {}.".format(data_id, resource))
      return False
```

Reply:
I see what you mean, I have updated the codes per suggestions